uk.co.compucorp.civicrm.tasksassignments/CRM/Activity/Form/ActivityView.php

```diff
@@ -46,9 +46,12 @@ public function preProcess() {
     $cid = CRM_Utils_Request::retrieve('cid', 'Positive', $this);
 
     // Check for required permissions, CRM-6264.
-    if ($activityId &&
-      !CRM_Activity_BAO_Activity::checkPermission($activityId, CRM_Core_Action::VIEW)
-    ) {
+    if ($activityId && (
+      !CRM_Activity_BAO_Activity::checkPermission($activityId, CRM_Core_Action::VIEW) &&
+      // @custom We add the line below to also check for permission defined in
```

Comment:
now since the custom code is clear to me .. I think you probably do not need to override this page .. I think you can use this hook https://wiki.civicrm.org/confluence/display/CRMDOC/hook_civicrm_preProcess